packages/maskbook/src/database/Persona/Persona.db.ts

```diff
@@ -477,6 +501,86 @@ export async function deleteProfileDB(id: ProfileIdentifier, t: ProfileTransacti
     MaskMessage.events.profilesChanged.sendToAll([{ reason: 'delete', of: id }])
 }
 
+/**
+ * Create a new Relation
+ */
+export async function createRelationDB(
+    record: Omit<RelationRecord, 'network'>,
+    t: RelationTransaction<'readwrite'>,
+): Promise<void> {
+    await t.objectStore('relations').add(relationRecordToDB(record))
+    MaskMessage.events.relationsChanged.sendToAll([{ of: record.profile, reason: 'update', favor: record.favor }])
+}
+
+/**
+ * Query relations by paged
+ */
+export async function queryRelationsPagedDB(
+    linked: PersonaIdentifier,
+    options: {
+        network: string
+        after?: RelationRecord
+    },
+    count: number,
+) {
+    const t = createTransaction(await db(), 'readonly')('relations')
+    let firstRecord = true
+
+    const data: RelationRecord[] = []
+
+    for await (const cursor of t.objectStore('relations').index('linked, profile, favor').iterate()) {
+        if (cursor.value.linked !== linked.toText()) continue
+        if (
+            firstRecord &&
+            options.after &&
+            options.after.linked.toText() !== cursor?.value.linked &&
+            options.after.profile.toText() !== cursor?.value.profile
+        ) {
+            cursor.continue([options.after.linked, options.after.profile, options.after.favor].join(', '))
```

Comment:
Why did you join the index? I guess the index is an array

Reply:
yes

Reply:
So I think the parameter of continue should be an array